addons/web/tests/test_captcha.py

```diff
@@ -0,0 +1,40 @@
+from contextlib import contextmanager
+from unittest.mock import patch
+
+from odoo import http
+from odoo.tests import tagged
+from odoo.tests.common import HttpCase
+
+
+@tagged('post_install', '-at_install')
+class TestCaptcha(HttpCase):
+    def setUp(self):
+        super().setUp()
+        self.authenticate(None, None)
+
+    @contextmanager
+    def patch_captcha_valid(self, validity):
+        def _verify_request_recaptcha_token(self, captcha):
+            return validity
+        with patch.object(self.env.registry['ir.http'], '_verify_request_recaptcha_token', _verify_request_recaptcha_token):
+            yield
+
+    def test_post_valid(self):
+        with self.patch_captcha_valid(True):
+            res = self.url_open('/web/login', data={'csrf_token': http.Request.csrf_token(self), 'login': '_', 'password': '_'})
+            self.assertEqual(res.status_code, 200)
```

Comment:
```suggestion
            res.raise_for_status()
            self.assertEqual(res.status_code, 200)
```
in case of an assertion failure, the error message would be `400` is not equal `200`, which is not super explanatory. Using `raise_for_status()` first, we get the body of the response, which hold the UserError message :)